readthedocs/organizations/views/public.py

```diff
@@ -8,21 +8,27 @@
 
 from readthedocs.organizations.models import Team
 from readthedocs.organizations.views.base import (
-    CheckOrganizationsEnabled,
     OrganizationMixin,
     OrganizationTeamMemberView,
     OrganizationTeamView,
     OrganizationView,
 )
+from readthedocs.organizations.views.decorators import (
+    redirect_if_organizations_disabled,
+)
 from readthedocs.projects.models import Project
 
 log = structlog.get_logger(__name__)
 
 
-class OrganizationTemplateView(CheckOrganizationsEnabled, TemplateView):
+class OrganizationTemplateView(TemplateView):
 
     """Wrapper around `TemplateView` to check if organizations are enabled."""
 
+    @redirect_if_organizations_disabled
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
```

Comment:
@ericholscher for instance here, there is only the `redirect_if_organizations_disabled` active, that's why it wasn't possible to expand the `dispatch` method of the former `CheckOrganizationsEnabled` without adding additional mixin classes or conditional statements and such.

Authentication is hard, it gets harder by adding inheritance complexities. For instance, an additional superclass with a `dispatch` method could silently disable intended logic.